### ISSUE TITLE
CurlResult current header regex

### ISSUE THREAD

**Issue by nupplaphil** [friendica/friendica#5886]
https://github.com/friendica/friendica/blob/71a60c5f66bd1f3f71e908241db711f5e4e1088d/src/Network/CurlResult.php#L122
I get errors in my test at the line showed above.

The error is gone when changing
```php
/^HTTP\/[1-2].+? [1-5][0-9][0-9]/
```
to
```php
/^HTTP\/[1-2].+?[1-5][0-9][0-9]/
``` 

I searched in the commits and found that exactly this regex is used for years. So before making any (maybe wrong) conclusions, I want to ask if somebody can see, if the whitespace does or doesn't matter (normaly in regex everything matters ;-) ).

I also made a online test:
works: https://www.phpliveregex.com/p/pxt
doesn't work: https://www.phpliveregex.com/p/pxu

What do you say?

**Comment by MrPetovan** [comment]
`+?` means `at least 1, as few as possible`

Which means that if there's no minor version, the pattern with the space won't be able to match anything between the major version number and the space in the pattern and will fail.

This regular expression has been working because there's always been a minor version even with `HTTP/1.0`.

**Comment by nupplaphil** [comment]
Sounds like it is a bug for every request which doesn't respond a minor version?

I tested the regex with some possible/faulty response headers
https://www.phpliveregex.com/p/pxv

As far as I can see, I can fix it without side effect.

**Comment by MrPetovan** [comment]
Why restrict to version 1-2 and return coces? Why not using `/^HTTP\/.+? \d+/` instead?

**Comment by nupplaphil** [comment]
hm .. So we will parse the header in every case?
I thought maybe the RegEx was to avoid loading "wrong" headers.

**Comment by MrPetovan** [comment]
No, the space is a constant, with your Regex you can have `HTTP/22` and it would work. With my Regex it would need to be at least `HTTP/2 2`. I added back the space but I removed the `[1-2]` which allows the `.+?` to capture the whole version, no matter how it is formatted.

**Comment by nupplaphil** [comment]
Ah, you're right!

**Comment by nupplaphil** [comment]
fixed with #5888
